### PR TITLE
Collection creation refactor

### PR DESCRIPTION
**Pull request by dangalan**
- Refactor + simplified collection creation hook
- Switched from drawer to modal
- Made modal sizing responsive

https://github.com/MutualAidNYC/services-lists/assets/54749942/9c8f465c-3c75-4dee-b63e-69985fac76f8

### CODE REVIEW

File: components/CreateCollectionModal.tsx
Comment (dangalan) [inline components/CreateCollectionModal.tsx:1]: Code mostly copied from `CreateListDrawer.tsx1; added placeholder text, required field message, and error message for trying to add the same resource multiple times

File: components/forms/FormControl.tsx
Comment (dangalan) [inline components/forms/FormControl.tsx:1]: Could prob be refactored further in a future PR but pulling out some of the common `FormControl` code into its own component

**Comment by gnoahp** [comment]
Mostly looks good to me (not sure I know enough to really review), but I think you lost some of the homepage responsive stuff (see comment above).

File: pages/index.tsx [hunk 1]
```diff
@@ -161,11 +149,7 @@ export const HomePage: NextPage = () => {
         <meta name="image" content="/manyc_logo.png" />
         <link rel="icon" href="/icon.ico" />
       </Head>
-      <CreateListProvider value={createListHandler}>
-        <CreateListAlert selectedService={selectedResource} />
-        <CreateListDrawer />
-      </CreateListProvider>
-      <Box px={{ base: '10%', lg: '112px' }} py="96px">
+      <Box px="112px" py="96px">
```
[the remark this answers is not in the document]
Comment (dangalan) [reply pages/index.tsx:152]: oops, good catch! I must have just accepted all my current changes in rebasing this file 😬

**Comment by dangalan** [comment]
> Mostly looks good to me (not sure I know enough to really review), but I think you lost some of the homepage responsive stuff (see comment above).

I think I'm the only one the worked on this flow before so idk if Nazmus would know ~that~ much more of the business logic than you - if the code looks good to you I say go ahead and approve it!

Happy to walk thru the code before you approve too if that's helpful.

**Review by gnoahp — APPROVED** [review]
lgtm

**Pull request merged**